src/ripple/app/ledger/impl/LedgerConsensusImp.cpp

```diff
@@ -981,11 +981,19 @@ void LedgerConsensusImp::accept (std::shared_ptr<SHAMap> set)
         auto buildLCL = std::make_shared<Ledger>(
             *mPreviousLedger,
             app_.timeKeeper().closeTime());
-        auto constexpr v2_ledger_seq_switch = 40'000'000;
-        if (buildLCL->info().seq > v2_ledger_seq_switch &&
-           !buildLCL->stateMap().is_v2())
+        auto const v2_enabled = buildLCL->rules().enabled(featureSHAMapV2,
+                                                       app_.config().features);
+        auto v2_transition = false;
+        if (v2_enabled && !buildLCL->stateMap().is_v2())
         {
-            buildLCL->make_v2();
+            int txCount = 0;
+            for (auto& txn : *set)
+                ++txCount;
+            if (txCount > 10 && mHaveCorrectLCL)
```

Comment:
Can you explain? Casual inspection seems to imply my version is  mfunctionally identical to the original.